toolchain/check/testdata/facet/facet_assoc_const.carbon

```diff
@@ -140,15 +140,14 @@ fn F(T:! M & (L where .W = {}), a: T.W) {
   // the impl or `{}` from the facet type of T.
 
   let b: () = G(T, a);
-  // CHECK:STDERR: fail_two_different_combined_from_final_impl_and_facet.carbon:[[@LINE+10]]:15: error: cannot implicitly convert expression of type `()` to `{}` [ConversionFailure]
-  // CHECK:STDERR:   let c: {} = H(T, a);
-  // CHECK:STDERR:               ^~~~~~~
-  // CHECK:STDERR: fail_two_different_combined_from_final_impl_and_facet.carbon:[[@LINE+7]]:15: note: type `()` does not implement interface `Core.ImplicitAs({})` [MissingImplInMemberAccessNote]
+  // FIXME: This diagnostic sucks. Can we make the facet type's value take precidence over final, since that's what is written in the code and more likely to show up in diagnostics?
```

Comment:
FWIW, I think the ideal thing would be to diagnose the type of `T` itself, given that its `.W = {}` constraint would substitute to `() = {}`. But I guess we'll need to decide whether we'd consider this function to be valid if it didn't call `H`.

Reply:
I thought we had decided that the function needs an error somewhere but we preferred it to be on one of the uses of `.W` as long as both would see the same value for `.W`. I can persue diagnosing if you think that would be better though. Producing a better diagnostic here is going to be hard, I guess we'd need to print the associated constants from every `impl` reachable from `T` as well. I wouldn't want to just try combine them all into one facet type - if we were going that way I'd prefer to just diagnose `T` then since constructing that combined thing is invalid.

I have changed this FIXME to a TODO for now, but want to revisit. Adding a task to my task list :)